### PR TITLE
Comments for the process of reconstruction

### PR DESCRIPTION
**Pull request by UladBohdan**

### CODE REVIEW

**Comment by UladBohdan** [comment]
That was done mostly for learning purposes: the understanding of how reconstruction works and where code is located is now much more clear.

(feeling myself like after a long lesson of English writing)

File: src/optimization.cpp [hunk 1]
```diff
@@ -320,6 +327,8 @@ void BundleAdjuster::FillPoints(
                 continue;
             }
 
+            // Some 3d points has been filtered out as we we have nothing to do with them.
```
Comment (Drapegnik) [inline src/optimization.cpp:330]: misprint 😂

File: src/processor/incremental_mapper.cpp [hunk 2]
```diff
@@ -190,20 +210,30 @@ bool IncrementalMapper::RegisterInitialImagePair(const Options& options,
     Image& image2 = reconstruction_->Image(image_id2);
     const Camera& camera2 = reconstruction_->Camera(image2.CameraId());
 
+    // Ok so now er have image1 and image2 which are initial images.
```
Comment (Drapegnik) [inline src/processor/incremental_mapper.cpp:213]: misprint

File: src/processor/incremental_mapper_controller.cpp [hunk 3]
```diff
@@ -347,71 +383,93 @@ void IncrementalMapperController::run() {
...
-                              " (" + std::to_string(reconstruction.NumRegImages() + 1) +
-                              ")");
+                              " (" + std::to_string(reconstruction.NumRegImages() + 1) + ")");
 
                 std::cout << "  => Image sees " << next_image.NumVisiblePoints3D()
-                << " / " << next_image.NumObservations() << " points."
-                << std::endl;
+                          << " / " << next_image.NumObservations() << " points." << std::endl;
 
+                // RegusterNextImage() is really huge. And compilicated. Lot's of things are happening inside.
```
Comment (Drapegnik) [inline src/processor/incremental_mapper_controller.cpp:409]: misprint

File: src/processor/incremental_mapper_controller.cpp [hunk 4]
```diff
@@ -347,71 +383,93 @@ void IncrementalMapperController::run() {
...
                         mapper_options.IncrementalMapperOptions(), next_image_id);
 
                 if (reg_next_success) {
+                    // So image is successfully added. That's not the end. We have to suffer more.
+
+                    // Greeting IncrementalTriangulator!
+                    // As for now let's assume it makes the world a better place.
                     TriangulateImage(mapper_options, next_image, &mapper);
+
+                    // I thought we're close to the end - but here we're bundle-adjusting again! What a surprise!
```
Comment (Drapegnik) [inline src/processor/incremental_mapper_controller.cpp:424]: > but here we're bundle-adjusting again! What a surprise!
😆 😆 😆

File: src/processor/incremental_mapper_controller.cpp [hunk 5]
```diff
@@ -347,71 +383,93 @@ void IncrementalMapperController::run() {
...
-                        mapper_options.ba_global_points_ratio * prev_num_points ||
-                        reconstruction.NumPoints3D() >=
-                        mapper_options.ba_global_points_freq + prev_num_points) {
+                    if (reconstruction.NumRegImages() >= mapper_options.ba_global_images_ratio * prev_num_reg_images ||
+                        reconstruction.NumRegImages() >= mapper_options.ba_global_images_freq + prev_num_reg_images ||
+                        reconstruction.NumPoints3D() >= mapper_options.ba_global_points_ratio * prev_num_points ||
+                        reconstruction.NumPoints3D() >= mapper_options.ba_global_points_freq + prev_num_points) {
+                        // Not the easiest condition. So if it's true we're doing some magic with tracks (merging
+                        // some of them, completing them) and running bundle adjustment again. Just in case, I guess.
+                        // Because we can.
```
Comment (Drapegnik) [inline src/processor/incremental_mapper_controller.cpp:435]: 👍

**Comment by UladBohdan** [comment]
fixed.

**Pull request merged**